openmdao/components/mux_comp.py

```diff
@@ -0,0 +1,137 @@
+"""Definition of the Mux Component."""
+
+
+from six import iteritems
+
+import numpy as np
+
+from openmdao.core.explicitcomponent import ExplicitComponent
+
+
+class MuxComp(ExplicitComponent):
+    """
+    Mux one or more inputs along a given axis.
+
+    Attributes
+    ----------
+    _vars : dict
+        Container mapping name of variables to be muxed with additional data.
+    _input_names : dict
+        Container mapping name of variables to be muxed with associated inputs.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Instantiate MuxComp and populate private members.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Arguments to be passed to the component initialization method.
+        """
+        super(MuxComp, self).__init__(**kwargs)
+
+        self._vars = {}
+        self._input_names = {}
+
+    def initialize(self):
+        """
+        Declare options.
+        """
+        self.options.declare('vec_size', types=int, default=2,
+                             desc='The number of elements to be combined into an output.')
+
+    def add_var(self, name, val=1.0, shape=None, units=None, desc='', axis=0):
+        """
+        Add an output variable to be muxed, and all associated input variables.
+
+        Parameters
+        ----------
+        name : str
+            name of the variable in this component's namespace.
+        val : float or list or tuple or ndarray or Iterable
+            The initial value of the variable being added in user-defined units.
+            Default is 1.0.
+        shape : int or tuple or list or None
+            Shape of the input variables to be muxed, only required if val is not an array.
+            Default is None.
+        units : str or None
+            Units in which this input variable will be provided to the component
+            during execution. Default is None, which means it is unitless.
+        desc : str
```

Comment:
Confused on wording of some attributes, including axis and units. May have been cut and paste from Demux.